From 497f2682957d4cdea7f3d112744db1ed16a3e231 Mon Sep 17 00:00:00 2001
From: Ankit Gupta <agupta@cyngn.com>
Date: Tue, 14 Jun 2016 13:49:47 -0700
Subject: [PATCH 15/16] MediaPicker: Check for NPE

make sure view has been created before asking for
its visibility.

Change-Id: I0f0c36f8edf83c518bf07ef05399eda5d6b0f78a
Issue-id: FEIJ-1135
---
 .../android/messaging/ui/mediapicker/AudioMediaChooser.java   | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/com/android/messaging/ui/mediapicker/AudioMediaChooser.java b/src/com/android/messaging/ui/mediapicker/AudioMediaChooser.java
index 5d79293..3553f10 100644
--- a/src/com/android/messaging/ui/mediapicker/AudioMediaChooser.java
+++ b/src/com/android/messaging/ui/mediapicker/AudioMediaChooser.java
@@ -123,6 +123,10 @@ class AudioMediaChooser extends MediaChooser implements
             final int requestCode, final String permissions[], final int[] grantResults) {
         if (requestCode == MediaPicker.RECORD_AUDIO_PERMISSION_REQUEST_CODE) {
             final boolean permissionGranted = grantResults[0] == PackageManager.PERMISSION_GRANTED;
+            // onRequestPermissionsResult can sometimes get called before createView().
+            if (mEnabledView == null) {
+                return;
+            }
             mEnabledView.setVisibility(permissionGranted ? View.VISIBLE : View.GONE);
             mMissingPermissionView.setVisibility(permissionGranted ? View.GONE : View.VISIBLE);
         }
-- 
2.17.1

